### PR TITLE
fix(ssri): Return `Promise` type by default

### PR DESCRIPTION
**Pull request by ExE-Boss**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/zkat/ssri/blob/v6.0.1/README.md#api>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

---

Since I’ve completely rewritten the tests to use `$ExpectType`, I’ve ended up reformatting them to use tabs.

I’ve also added union returning overloads for `parse(…)` and `fromHex(…)` until Microsoft/TypeScript#14107 is resolved.

### CODE REVIEW

**Comment by huan086** [comment]
Could you keep the changes to the minimum by using 4 spaces instead of tabs? Majority of the types uses spaces. I found tabs in 22% of the packages, others uses spaces.

For comma dangle, I think it makes sense only for multi-line arguments. Changes on the same line will always show up in diff even with comma dangle.

**Comment by ExE-Boss** [comment]
You can use the `?w=1` query parameter to hide whitespace‑only changes in the diff: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/34892/files?w=1

**Review by huan086 — APPROVED** [review]

File: types/ssri/index.d.ts [hunk 1]
```diff
@@ -1,126 +1,181 @@
...
+): IntegrityMap | Hash;
 
 export function stringify(
-    obj: string | IntegrityLike | HashLike,
-    opts?: { strict?: boolean, sep?: string }
+	obj: string | IntegrityLike | HashLike,
+	opts?: { strict?: boolean, sep?: string, },
 ): string;
 
 export function fromHex(
```
Comment (PranavSenthilnathan) [inline types/ssri/index.d.ts]: Another option is to just have one function like:
```
export function fromHex<T extends boolean = false>(
  hexDigest: string, 
  algorithm: string, 
  opts?: { 
    single?: T, 
    string?: boolean, 
    options?: ReadonlyArray<string>
  }
): T extends true ? Hash : IntegrityMap;
```

**Review by PranavSenthilnathan — CHANGES_REQUESTED** [review]

File: types/ssri/index.d.ts (hunk 1 above)
Comment (ExE-Boss) [reply types/ssri/index.d.ts]: That would bump the required TypeScript version to 2.8.

File: types/ssri/tslint.json [hunk 2]
```diff
@@ -1 +1,9 @@
-{ "extends": "dtslint/dt.json" }
+{
+	"extends": "dtslint/dt.json",
+	"rules": {
```
[the remark this answers is not in the document]
Comment (ExE-Boss) [reply types/ssri/tslint.json]: @sandersn has approved me doing this in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/33478#issuecomment-470728638.

File: types/ssri/index.d.ts (hunk 1 above)
Comment (PranavSenthilnathan) [reply types/ssri/index.d.ts]: The TypeScript version should really be specified in the top of the d.ts file if people are going to be relying on a fixed version.

**Review by PranavSenthilnathan — APPROVED** [review]

**Pull request merged**